src/odemis/util/units.py

```diff
@@ -366,6 +371,9 @@ def readable_time(seconds, full=True):
     if msec:
         result.append(u"%d ms" % msec)
 
+    if usec:
```

Comment:
Was it a deliberate choice to not include a full word option for the ms and us? For the rest it looks good!